test/jdk/java/foreign/TestByteBuffer.java

```diff
@@ -569,4 +599,21 @@ static Object defaultValue(Class<?> c) {
             return null;
         }
     }
+
+    @DataProvider(name = "bufferSources")
+    public static Object[][] bufferSources() {
+        Predicate<MemorySegment> heapTest = segment -> segment instanceof HeapMemorySegment;
+        Predicate<MemorySegment> nativeTest = segment -> segment instanceof NativeMemorySegment;
+        Predicate<MemorySegment> mappedTest = segment -> segment instanceof MappedMemorySegment;
+        try (FileChannel channel = FileChannel.open(tempPath)) {
+            return new Object[][]{
+                    { ByteBuffer.wrap(new byte[256]), heapTest },
+                    { ByteBuffer.allocate(256), heapTest },
+                    { ByteBuffer.allocateDirect(256), nativeTest },
+                    { channel.map(FileChannel.MapMode.READ_ONLY, 0L, 256), mappedTest }
```

Comment:
Isn't the file channel closed after return? I am suppose this is OK as we don't access it and probably intentional here.

What I am curious about, is what suppose to happen to the MS after this channel is closed? Since this is from MappedByteBuffer, I suppose it inherits the unspecified behavior.

Reply:
My understanding is that lifecycle of a mapped buffer is not tied to that of the channel where it came from. The API for `FileChannel::map` says:

```
 A mapping, once established, is not dependent upon the file channel that was used to create it. Closing the channel, in particular, has no effect upon the validity of the mapping. 
```